feat: Rework quick casting

ShadowCat117 opened the pull request:
I'm not a macro user myself but just based on a quick comparison between the old and new version of quick casting this one does seem to be a lot better, for starters I was unable to make it cast the wrong spell when just holding down a key.

Safe casting has been removed, I spent quite a bit of time trying to get it to work nicely by removing spells from the queue if the current in progress spell is compatible but quite often it would just end up casting the wrong spells and leaving an extra click in the next spell queue so as nice as it would be to have this work, I've just opted to remove it. Maybe it can be revisited in the future.

Have also removed the swapping back to the item if you changed item whilst a cast was in progress, unless I was doing this wrong it didn't seem to work anyway so nothing lost, it now just clears the queue on item swap.

There's 3 new configs, a left click tick delay, just works like the right click one but for left clicks. An option to block left/right clicks whilst there is a queue in progress just to stop wrong casts and a config to ignore weapon check, this would be useful if we get an issue with weapons not being detected correctly again.

I would encourage others to test this out and try to find any issues with the default settings, lowering the tick delay will probably have some wrong casts but 2 did seem stable for me when I tried it.

ShadowCat117 committed:
feat: Rework quick casting

kristofbolyai reviewed (approved):
Code looks good, from my limited testing, the delays feel fine too.

ShadowCat117 committed:
chore: Remove in progress cancel message

ShadowCat117 committed:
chore: Make keybinds not first press only

First press only doesn't work anyway, but in case it's fixed this doesn't seem to be illegal as spellcaster mod works the same

ShadowCat117 committed:
ci: spotless formatting
